code-of-conduct.md

```diff
@@ -0,0 +1,84 @@
+# Contributor Covenant Code of Conduct
+
+## Our Pledge
+
+We as members, contributors, and leaders pledge to make participation in our community a harassment-free experience for everyone, regardless of age, body size, visible or invisible disability, ethnicity, sex characteristics, gender identity and expression, level of experience, education, socio-economic status, nationality, personal appearance, race, religion, or sexual identity and orientation.
+
+We pledge to act and interact in ways that contribute to an open, welcoming, diverse, inclusive, and healthy community.
+
+## Our Standards
+
+Examples of behavior that contributes to a positive environment for our community include:
+
+* Demonstrating empathy and kindness toward other people
+* Being respectful of differing opinions, viewpoints, and experiences
+* Giving and gracefully accepting constructive feedback
+* Accepting responsibility and apologizing to those affected by our mistakes, and learning from the experience
+* Focusing on what is best not just for us as individuals, but for the overall community
+
+Examples of unacceptable behavior include:
+
+* The use of sexualized language or imagery, and sexual attention or
+  advances of any kind
+* Trolling, insulting or derogatory comments, and personal or political attacks
+* Public or private harassment
+* Publishing others' private information, such as a physical or email
+  address, without their explicit permission
+* Other conduct which could reasonably be considered inappropriate in a
+  professional setting
+
+## Enforcement Responsibilities
+
+Community leaders are responsible for clarifying and enforcing our standards of acceptable behavior and will take appropriate and fair corrective action in response to any behavior that they deem inappropriate, threatening, offensive, or harmful.
+
+Community leaders have the right and responsibility to remove, edit, or reject comments, commits, code, wiki edits, issues, and other contributions that are not aligned to this Code of Conduct, and will communicate reasons for moderation decisions when appropriate.
+
+## Scope
+
+This Code of Conduct applies within all community spaces, and also applies when an individual is officially representing the community in public spaces. Examples of representing our community include using an official e-mail address, posting via an official social media account, or acting as an appointed representative at an online or offline event.
+
+## Enforcement
+
+Instances of abusive, harassing, or otherwise unacceptable behavior may be reported to the community leaders responsible for enforcement `[at ... | as set forth in the repository's ... file | ... ]`. All complaints will be reviewed and investigated promptly and fairly.
```

Comment:
I agree in principle. Not sure we have that many people right now...

Reply:
I completely agree that CoC violations team could be different than the SC, but if so, who should be on it? I guess we as the SC should make a decision on this.

@adrinjalali , @romanlutz , @hildeweerts : I'm personally leaning towards just starting off with the SC, but if you'd rather this be separate from the SC, we could try to ask some community members (around two?) to be responsible for this. Thoughts?

If there is no strong sentiment either way, I'd just default to:
```
Instances ... may be reported to any of the Steering Committee members by e-mail.
```

Reply:
I like the idea to be able to report to individuals rather than the entire COC/SC/maintainer team, so people can report to whoever they feel most comfortable with. Possibly include all maintainers instead of just the SC?

If it also includes reinforcement, I honestly don’t really know what that would/should look like, so I find this a bit difficult to form an opinion on.

Reply:
Being on the CoC team is really hard when a report comes in. It also takes a significant amount of cognitive and emotional load. I'm happy for anybody who wants to be on the CoC team, to be on it, but I don't think it's reasonable to expect everybody on the SC team, or the maintainer team, to be on it. I don't mind it if the CoC team is initially the same as the SC team. I'd suggest naming those people instead of saying the SC team (asking them beforehand if they want to be on it of course), and have users/contributors/folks report to any individual on that team as they wish. The team then takes care of it.

Reply:
Looking at most other projects, the standard mechanism is to just e-mail a dedicated to code-of-conduct e-mail address. Shall we use something like fairlearn-conduct at gmail (or some other e-mail provider?). I'm not sure how many e-mails at python.org we can ask to own :-)